intellij-plugin-structure/structure-youtrack/src/main/kotlin/com/jetbrains/plugin/structure/youtrack/YouTrackPluginManager.kt

```diff
@@ -0,0 +1,114 @@
+package com.jetbrains.plugin.structure.youtrack
+
+import com.fasterxml.jackson.module.kotlin.jacksonObjectMapper
+import com.jetbrains.plugin.structure.base.decompress.DecompressorSizeLimitExceededException
+import com.jetbrains.plugin.structure.base.plugin.*
+import com.jetbrains.plugin.structure.base.problems.*
+import com.jetbrains.plugin.structure.base.utils.*
+import com.jetbrains.plugin.structure.youtrack.bean.YouTrackAppManifest
+import com.jetbrains.plugin.structure.youtrack.problems.createIncorrectYouTrackPluginFileError
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import java.nio.file.Files
+import java.nio.file.Path
+import java.nio.file.Paths
+
+class YouTrackPluginManager private constructor(private val extractDirectory: Path) : PluginManager<YouTrackPlugin> {
+
+  companion object {
+    const val DESCRIPTOR_NAME = "manifest.json"
+
+    private val LOG: Logger = LoggerFactory.getLogger(YouTrackPluginManager::class.java)
+
+    fun createManager(
+      extractDirectory: Path = Paths.get(Settings.EXTRACT_DIRECTORY.get())
+    ): YouTrackPluginManager {
+      extractDirectory.createDir()
+      return YouTrackPluginManager(extractDirectory)
+    }
+  }
+
+  override fun createPlugin(pluginFile: Path): PluginCreationResult<YouTrackPlugin> {
+    require(pluginFile.exists()) { "Plugin file $pluginFile does not exist" }
+    return when {
+      pluginFile.isDirectory -> loadPluginInfoFromDirectory(pluginFile)
+      pluginFile.isZip() -> loadDescriptorFromZip(pluginFile)
+      else -> PluginCreationFail(createIncorrectYouTrackPluginFileError(pluginFile.simpleName))
+    }
+  }
+
+
+  private fun loadDescriptorFromZip(pluginFile: Path): PluginCreationResult<YouTrackPlugin> {
+    val sizeLimit = Settings.YOUTRACK_PLUGIN_SIZE_LIMIT.getAsLong()
+    if (Files.size(pluginFile) > sizeLimit) {
+      return PluginCreationFail(PluginFileSizeIsTooLarge(sizeLimit))
+    }
+
+    val tempDirectory = Files.createTempDirectory(extractDirectory, "plugin_")
+    return try {
+      extractZip(pluginFile, tempDirectory, sizeLimit)
+      loadPluginInfoFromDirectory(tempDirectory)
+    } catch (e: DecompressorSizeLimitExceededException) {
+      return PluginCreationFail(PluginFileSizeIsTooLarge(e.sizeLimit))
+    } finally {
+      tempDirectory.deleteLogged()
+    }
+  }
+
+  private fun loadPluginInfoFromDirectory(pluginDirectory: Path): PluginCreationResult<YouTrackPlugin> {
+    val manifestFile = pluginDirectory.resolve(DESCRIPTOR_NAME)
+    if (!manifestFile.exists()) {
+      return PluginCreationFail(PluginDescriptorIsNotFound(DESCRIPTOR_NAME))
+    }
+    val manifestContent = manifestFile.readText()
+    val mapper = jacksonObjectMapper()
```

Comment:
A field variable? ObjectMapper is thread-safe.

Reply:
I don't see any issue here, field variables are free and just increase readablity even they used only once.
But I fixed it as you asked :)

Please note that `EduPluginManager`, `HubPluginManager`, `TeamcityPluginManager` have the same.